repository-data/webfiles/src/main/resources/site/freemarker/common/macro/sections/tableauSection.ftl

```diff
@@ -15,20 +15,51 @@
         <div id="${divId}"></div>
 
         <script type="text/javascript">
-            window.addEventListener('load', initViz);
+            var viz;
 
-            function initViz() {
-                var containerDiv = document.getElementById("${divId}");
+            function throttleViz(containerDiv) {
+                <#if section.throttlingLocation??>
+                    <#local tableauThrottleSize="uk.nhs.digital.freemarker.tableau.RemoteThrottleSizeFromUrl"?new() />
+                    <#local remote = tableauThrottleSize(section.throttlingLocation) />
+                    var attemptProbability = ${remote.size};
```

Comment:
Are we guaranteed a number here? If not, the calculation on line 30 might be a bit wacky, as JS will try to cooerce to something comparable. It might be worth testing with `isNaN`: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/isNaN or similar

Reply:
Good thinking.